util/topgen/templates/chiplevel.sv.tpl

```diff
@@ -1222,6 +1225,36 @@ module chip_${top["name"]}_${target["name"]} #(
       {soc_proxy_pkg::NumFatalExternalAlerts{soc_proxy_pkg::SOC_ALERT_REQ_DEFAULT}};
   assign soc_recov_alert_req =
       {soc_proxy_pkg::NumRecovExternalAlerts{soc_proxy_pkg::SOC_ALERT_REQ_DEFAULT}};
+  // The logic below is a is a loop back path
+  // It listens to the internal reset request generated by the power manager
+  // Translates this request into an expanded pulse (modeled by the counter)
+  // The stretched pulse is fed back to top_darjeeling as external async soc reset request
+  // There is opportunity to move this logic as part of the DV test bench
+  // Also need the ability to asynchronously assert the external reset (pin level)
+  // without any internal request
+  logic  internal_request_d, internal_request_q;
```

Comment:
We need to create an issue to track this.

Reply:
https://github.com/lowRISC/opentitan/issues/22710